test/test_utils.py

```diff
@@ -736,5 +737,17 @@ def test_exit(self):
             do_command(args, HERE)
 
 
+class FlattenTest(unittest.TestCase):
+    def test_good(self):
+        array_3d = np.empty((200, 4, 4))
+        flattened = flatten_chains(array_3d)
+        self.assertEqual(flattened.shape, (800, 4))
```

Comment:
I think it is important not just to test the shape, but also that the stacking behavior is in the desired (column-major?) order. This should be rather simple to test by using something other than `np.empty` like a few concatenated `np.arange` calls, and then assertions like 'element 400 is equal to element 0' and 'element 2 is not equal to element 0'

Reply:
agree - added more sets